great_expectations/data_context/store/expectations_store.py

```diff
@@ -341,6 +341,12 @@ def deserialize(self, value):  # type: ignore[explicit-override] # FIXME
         else:
             raise TypeError(f"Cannot deserialize value of unknown type: {type(value)}")  # noqa: TRY003
 
+    def deserialize_suite_dict(self, suite_dict: dict) -> ExpectationSuite:
+        suite = ExpectationSuite(**suite_dict)
+        if suite._include_rendered_content:
+            suite.render()
+        return suite
```

Comment:
Small step towards moving this logic here - if I override `deserialize`, things seem to break? This seems like a viable solution for now but I don't like having multiple "deserialize" methods here